frontend/src/App.js

```diff
@@ -118,20 +122,28 @@ export default class App extends React.Component {
     });
   };
 
-  createDiagram = () => {
+  createDiagram = ({ diagramName }) => {
+    const { diagrams } = this.state;
     this.socket.send(
-      JSON.stringify({ message: "creatediagram", data: this.state.diagramName })
+      JSON.stringify({ message: "creatediagram", data: diagramName })
     );
     this.setState({
       diagramName: "",
-      diagrams: [...this.state.diagrams, this.state.diagramName],
+      diagrams: [...diagrams, diagramName],
     });
+    this.props.history.push("/");
```

Comment:
what does this do?

Reply:
don't we want to redirect to `/diagrams/diagramName`?

Reply:
I was thinking to go to the list of diagrams, but it does make sense to go to the diagram's page. I'll change it now.